validation/src/blockchain/eth_rpc/types.rs

```diff
@@ -101,36 +93,25 @@ impl FromStr for EthProofType {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-/// A proof of time on the blockchain
-pub struct TimeProof {
+/// A proof of time derived from state on the blockchain
+pub struct TimeProofResults {
     /// The timestamp the proof was recorded
     pub timestamp: u64,
     /// The root CID of the proof
     pub root_cid: Cid,
 }
 
-/*
-    Planning to implemented this trait for Hoku using something like:
-
-    #[derive(Clone, Debug, PartialEq, Eq)]
-    pub struct HokuTxProof {
-        cid: Cid,
-        index: u64,
-    }
-*/
 #[async_trait::async_trait]
-/// Ethereum RPC provider methods. This is a higher level type than the actual RPC calls neeed and
+/// Wrapper around blockchain RPC provider that can be used to query blockchain state for the
+/// relevant information (such as the timestamp) for the given transaction described by an
+/// `AnchorProof`. This is a higher level type than the actual RPC calls need and
 /// may wrap a multiple calls into a logical behavior of getting necessary information.
 pub trait ChainInclusion {
-    /// The input format needed to do the inclusion proof
-    type InclusionInput;
-
     /// Get the CAIP2 chain ID supported by this RPC provider
     fn chain_id(&self) -> &caip2::ChainId;
 
     /// Get the block chain transaction if it exists with the block timestamp information
-    async fn chain_inclusion_proof(&self, input: &Self::InclusionInput)
-        -> Result<TimeProof, Error>;
+    async fn chain_inclusion_proof(&self, input: &AnchorProof) -> Result<TimeProofResults, Error>;
```

Comment:
Why do we use the AnchorProof struct when it has more information than is needed by the implementations? With this API its not clear if the `chain_inclusion_proof` method is checking if the root cid returned matches the root cid of the input proof structure. 

This API is about only getting the proof result correct? In that case I think I prefer having the associated type InclusionInput that makes it explicit what information a given provider needs in order to get the proof result. 

Thoughts? Maybe I have missed the motivation behind this change.

Reply:
Maybe we do `type InclusionInput: From<AnchorProof>` so that callers can still use the anchor proof but then its clear that the provider itself is not validating the proof but instead just retrieving the information the proof references?

Reply:
Per discussion on discord, there's unfortunately no way to do this without leaking details of the different ChainInclusion implementations for different chains into the calling code.  Decided the lesser evil is taking the AnchorProof, even though its taking in more information than is needed, as it allows the EventValidation code to avoid any special logic that depends on what chain is being validated.